python/tvm/contrib/graph_runtime.py

```diff
@@ -176,3 +191,151 @@ def __getitem__(self, key):
+        # get shape
+        op_shape = graph.json_attr("shape")
+
+        # get dtype
+        op_dtype = graph.json_attr("dltype")
+
+        # get nodes
+        op_nodes = graph.index.nodes
+
+        # get node row ptr
+        n_rp = graph.index.entry_ptr
+
+        # get start times
+        op_start = []
+        for i in range(graph.index.num_nodes):
+            op_start.append(self._get_op_start_time(i))
+
+        # get end times
+        op_end = []
+        for i in range(graph.index.num_nodes):
+            op_end.append(self._get_op_end_time(i))
+
+        # get node sizes
+        op_size = []
+        for i in range(graph.index.num_nodes):
+            op_size.append(self._get_op_size(i))
+
+        return op_shape, op_dtype, op_nodes, n_rp, op_start, op_end, op_size
+
+    def get_profile_data(self, graph):
+
+        # get operation info
+        op_shape, op_dtype, op_nodes, n_rp, op_start, op_end, op_size = self.fetch_op(graph)
+
+        # create profile
+        op_profile = []
+        for i in range(graph.index.num_nodes):
+            # skip non-operation nodes
+            if op_nodes[i]["op"] == "null": continue
+            # add operation information
+            op_dict = {"OpName": op_nodes[i]["name"],
+                       "OpType": op_nodes[i]["op"],
+                       "StartTime": op_start[i],
+                       "EndTime": op_end[i],
+                       "Duration": op_end[i]-op_start[i],
+                       "FuncName": op_nodes[i]["attrs"]["func_name"],
+                       "NumInput": int(op_nodes[i]["attrs"]["num_inputs"]),
+                       "NumOutput": int(op_nodes[i]["attrs"]["num_outputs"])}
+            # add input information
+            inputs = []
+            for idx in range(int(op_nodes[i]["attrs"]["num_inputs"])):
+                n_entry = op_nodes[i]["inputs"][idx]
+                n_idx = n_rp[n_entry[0]]+n_entry[1]
+                input_attr = {"InDim": len(op_shape[n_idx]),
+                              "InShape": op_shape[n_idx],
+                              "InDType": op_dtype[n_idx],
+                              "InSize": op_size[n_idx]}
+                inputs.append({"Input_%d"%(idx+1):input_attr})
+            op_dict["Input"] = inputs
+            # add output information
+            outputs = []
+            for idx in range(int(op_nodes[i]["attrs"]["num_outputs"])):
+                n_idx = n_rp[i]+idx
+                output_attr = {"OutDim": len(op_shape[n_idx]),
+                               "OutShape": op_shape[n_idx],
+                               "OutDType": op_dtype[n_idx],
+                               "OutSize": op_size[n_idx]}
+                outputs.append({"Output_%d"%(idx+1):output_attr})
+            op_dict["Output"] = outputs
+            # update profile
+            op_profile.append(op_dict)
+
+        return op_profile
+
+    def get_profile_json(self, graph, process=""):
```

Comment:
It looks like there are lots of codes which can be shared between get_profile_data and get_profile_json. Maybe we can organize them more cleanly.